p2p/client.go

```diff
@@ -194,22 +209,46 @@ func (s *peer) GetSignaturesForDocument(ctx context.Context, identityService ide
 		if err != nil {
 			return centerrors.Wrap(err, "failed to convert to CentID")
 		}
-		id, err := identityService.LookupIdentityForID(collaboratorID)
-		if err != nil {
-			return centerrors.Wrap(err, "error fetching collaborator identity")
-		}
-
-		receiverPeer, err := s.getPeerID(id)
-		if err != nil {
-			log.Error(centerrors.Wrap(err, "failed to connect to target"))
-			continue
+		tc, err := s.config.GetTenant(collaboratorID[:])
+		if err == nil {
```

Comment:
I think the concept is good, although I would rather have this switch done way down the pipeline, so we do not have to copy this code elsewhere. Meaning when invoking `s.mes.sendMessage()` chose there if is local or remote, should be transparent to the business logic (as much as possible). Let's sit down tomorrow and draft out a solution.